tests/testthat/test-namespace.R

```diff
@@ -0,0 +1,17 @@
+test_that("is_s3_generic", {
+  func <- function(x) {
+    print(x)
+    UseMethod("func")
+  }
```

Comment:
Is there a valid use case for the other way around?
i.e. for
```
func <- function(x) {
  x <- UseMethod("func")
  common_postprocessing(x)
}
```

Reply:
let's try & keep close to the base definition unless there's a compelling reason? i'm also a bit hesitant for this change if it's not being suggested upstream as well

Reply:
The base definition checks the first call, so would find my example iinm. That's why I'm asking.

Reply:
isn't the call that's found `<-` though, not UseMethod?

Reply:
Ah, right. I wonder why base R looks for the first call instead of the last, then...

Reply:
I didn't see anything about isS3stdGeneric() on r-devel in a quick search... maybe worth bringing up there.

cc @moodymudskipper who identified the bug fix at root in #1161